### PR TITLE
HW 12

### PR DESCRIPTION
**Pull request by geo-tat**

### CODE REVIEW

File: src/main/java/com/andersen/hw/config/AppSpringConfig.java [hunk 1]
```diff
@@ -1,48 +1,15 @@
...
-        return new JdbcTemplate(dataSource());
-    }
 
     @Bean
-    public PlatformTransactionManager transactionManager() {
-        return new DataSourceTransactionManager(dataSource());
+    @ConditionalOnProperty(name = "custom.bean.enabled", havingValue = "true")
+    public String thisIsMyFirstConditionalBean() {
+        return "My Bean";
     }
```
Comment (AtticSiol) [inline src/main/java/com/andersen/hw/config/AppSpringConfig.java]: this is not a bean, it is a String.

when you create bean, then you need not just create a method with some condition, you also need to return this been, not a `String`:
```java
    @Bean
    @ConditionalOnProperty(name = "app.allowConditionalBean", havingValue = "true")
    public ThisIsMyFirstConditionalBean thisIsMyFirstConditionalBean() {
        return new ThisIsMyFirstConditionalBean();
    }
```

File: src/main/java/com/andersen/hw/controller/TicketController.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+import java.util.List;
+
+@RequestMapping("/tickets")
+@RestController
+@RequiredArgsConstructor
+public class TicketController {
+    private final TicketService ticketService;
+
+    @PostMapping
+    void createTicket(@RequestBody Ticket ticket) {
```
Comment (AtticSiol) [inline src/main/java/com/andersen/hw/controller/TicketController.java]: all methods in controller have to be with `public`. if you need to make them with other type of access modifier - you need to have a reason for that

Comment (geo-tat) [reply src/main/java/com/andersen/hw/controller/TicketController.java]: done

File: src/main/java/com/andersen/hw/config/AppSpringConfig.java (hunk 1 above)
Comment (geo-tat) [reply src/main/java/com/andersen/hw/config/AppSpringConfig.java]: done

**Review by AtticSiol — APPROVED** [review]

**Pull request merged**